Monorepo Prep

askvortsov1 opened the pull request.

askvortsov1 committed:
chore: flarum-cli audit infra --fix

askvortsov1 committed:
fix: allow path repos

Temporarily configure cwd while running composer commands so that path repositories can be resolved properly.

Fixes https://github.com/flarum/package-manager/issues/6

askvortsov1 committed:
chore: bump yarn lock

SychO9 commented on .gitignore:
```diff
@@ -1,5 +1,12 @@
+.DS_Store
+Thumbs.db
+tests/.phpunit.result.cache
+/tests/integration/tmp
```
```suggestion
/tests/tmp
```

SychO9 commented on composer.json:
```diff
@@ -22,7 +26,6 @@
         "composer/composer": "^2.0"
     },
     "require-dev": {
-        "flarum/core": "dev-master",
```
this is needed, in all bundled extension in fact

askvortsov1 replied:
In the monorepo, minimum stability dev + prefer stable + path repo is sufficient for installing from the local dev version.

SychO9 replied:
I see! neat

SychO9 commented on composer.json:
```diff
@@ -53,15 +73,14 @@
         ],
         "test:unit": "phpunit -c tests/phpunit.unit.xml",
         "test:integration": "phpunit -c tests/phpunit.integration.xml",
-        "test:setup": [
-            "@php tests/integration/setup.php",
-            "cd $FLARUM_TEST_TMP_DIR_LOCAL && composer install"
-        ]
+        "test:setup": "@php tests/integration/setup.php"
```
gotta keep this unchanged

SychO9 commented on .github/workflows/test.yml:
```diff
@@ -1,85 +0,0 @@
-          DB_PORT: 13306
-          DB_PASSWORD: root
-          DB_PREFIX: ${{ matrix.prefix }}
-          FLARUM_TEST_TMP_DIR_LOCAL: tests/tmp
```
I think we lose this with the new workflow file, the location for the tests needs to be here instead of inside `integration` I don't actually remember why, I think it was because the tests install stuff, so a vendor is filled in `tmp` which then messes with other tests because PHPUnit starts looking inside all of that, so I had to move it out of `integration`.

askvortsov1 commented:
@SychO9 it mostly works now, with the exception of the deprecation notices. Not sure why we see these here, but not in core though, I don't think we explicitly disable them in core?

SychO9 commented on tests/integration/SetupComposer.php:
```diff
@@ -17,7 +17,7 @@ class SetupComposer
     private $config = [
         'require' => [
-            'flarum/core' => '1.0.0',
+            'flarum/core' => '^1.0.0',
```
This shouldn't need to be changed, it's for testing purposes

SychO9 reviewed (commented):
it's because one of the dependencies of composer forces deprecations to be thrown, technically while running tests deprecations *should* always be displayed, but the PHP setup action doesn't do that by default.

Though right now (in Flarum 1.x) we need deprecations not to be displayed because we can't do anything about them with PHP 7 support.

Anyway, we might be able to just force hide the deprecations by also changing the ini setting from the code, or we can drop 8.1 from this extension's test suite.

SychO9 commented:
If we can specify the php versions to run backend tests with so that we don't run them on php8.1 for now.

I don't know why tests are failing on 7.4, can't look into it atm, but we can also temporarily disable the tests here so that we can move forward with monorepo

askvortsov1 committed:
chore: exclude PHP 8.1 tests

These prompt deprecation warnings that break CI. We need to fix those, but that's a problem for a later time.

askvortsov1 committed:
fix: suppress xdebug warnings
